validator/client/wait_for_activation.go

```diff
@@ -54,29 +57,50 @@ func (v *validator) internalWaitForActivation(ctx context.Context, accountsChang
 		return v.waitForAccountsChange(ctx, accountsChangedChan)
 	}
 
-	// Step 3: update validator statuses in cache.
-	if err := v.updateValidatorStatusCache(ctx, validatingKeys); err != nil {
+	// Step 3: setup stream
+	stream, err := v.validatorClient.WaitForActivation(ctx, &ethpb.ValidatorActivationRequest{
+		PublicKeys: bytesutil.FromBytes48Array(validatingKeys),
+	})
+	if err != nil {
 		return v.retryWaitForActivation(ctx, span, err, "Connection broken while waiting for activation. Reconnecting...", accountsChangedChan)
```

Comment:
I didn't revert this as it was just broken out into a new function, it's functionally the same as the old version

https://github.com/prysmaticlabs/prysm/pull/14514/files#diff-32b491778d92918628a4a67eca393aab2c40aeb5fd0045d20e1051daa9795101L80